apps/trackers/jira/query.py

```diff
@@ -267,3 +271,89 @@ def generate_additional_fields(self):
                     continue
 
                 self._query["fields"][PS_ADDITIONAL_FIELD_TO_JIRA[name]] = field_value
+
+    def generate_cc(self):
+        """
+        generate query for CC list
+        """
+
+        # Each instance of TrackerJiraQueryBuilder is used only once, but if ever used twice,
+        # always produce consistent query and comment.
+        self._comment = None
+
+        if self.tracker.external_system_id:
+            # Add CCs only on creation.
+            return
+
+        # NOTE That SFM2 for Jira tracker creation uses ps_module.component_overrides only for
+        #      generating Jira "components" field, which OSIDB doesn't do, but not for CC lists;
+        #      CC list creation is based solely on ps_component, not on bz_component.
+        #      Therefore AffectCCBuilder.ps2bz_component is not reused here.
+        #      TODO: Is this a bug?
+        # Parse BZ component
+        if self.ps_component and "/" in self.ps_component:
+            bz_component = self.ps_component.split("/")[-1]
+        else:
+            bz_component = self.ps_component
+
+        # embargoed value unused here
+        affect_cc_builder = AffectCCBuilder(
+            self.tracker.affects.first(), embargoed=None
```

Comment:
`AffectCCBuilder` was mean to be used for each affect and the results were mean to be merged. Why is it only used for the first (which is probably close to random) affect here?

Reply:
It seemed like a great idea for a reason absolutely eluding me now, and I swear I wasn't under influence of anything. Thanks for pointing this out, will fix, this is wrong as it is now.

Reply:
Affect-based lists are generated for each affect in the revised version.